### PR TITLE
Define warning threshold. #19

### LINKED ISSUE #15: Add warning thredshold on counters check
Thredshold = 10, warning threshold = 5

Counter >= 0 ==> OK
Counter >= 5 ==> Warning
Counter >= 10 ==> KO

### PR DESCRIPTION
**Pull request by ycrumeyrolle**
Refactor Threshold as abstract classes.
Fix #15 & #20

### CODE REVIEW

File: src/AspNetCore.HealthCheck.Abstractions/CeilingThresholdWatchSettings.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            : base(name, critical, frequency, tags, errorThreshold, warningThreshold)
+        {
+        }
+
+        public override bool ReachWarningThreshold(long value)
+        {
+            return value >= WarningThreshold;
+        }
+
+        public override bool ReachErrorThreshold(long value)
```
Comment (sdib) [inline src/AspNetCore.HealthCheck.Abstractions/CeilingThresholdWatchSettings.cs]: HasReachedErrorThreshold would be a better name in my opinion

File: src/AspNetCore.HealthCheck.Abstractions/CeilingThresholdWatchSettings.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+namespace AspNetCore.HealthCheck
+{
+    public class CeilingThresholdWatchSettings: ThresholdWatchSettings
+    {
+        public CeilingThresholdWatchSettings(string name, bool critical, int frequency, IEnumerable<string> tags, long errorThreshold, long warningThreshold)
+            : base(name, critical, frequency, tags, errorThreshold, warningThreshold)
+        {
+        }
+
+        public override bool ReachWarningThreshold(long value)
```
Comment (sdib) [inline src/AspNetCore.HealthCheck.Abstractions/CeilingThresholdWatchSettings.cs]: HasReachedWarningThreshold would be a better name in my opinion

File: src/AspNetCore.HealthCheck.Counter/CounterWatcher.cs [hunk 3]
```diff
@@ -12,28 +12,37 @@ public CounterWatcher(ICounterProvider counterProvider)
...
+                    });
+            }
+            else if (settings.ReachWarningThreshold(counter.Value))
+            {
+                context.Warn(
                     properties: new Dictionary<string, object>
                     {
                         { "counter", counter.Value },
-                        { "threshold" , settings.Threshold }
+                        { "threshold" , settings.ErrorThreshold }
```
Comment (sdib) [inline src/AspNetCore.HealthCheck.Counter/CounterWatcher.cs]: shouldn't be settings.WarningThreshold ?

**Review by sdib — CHANGES_REQUESTED** [review]

File: src/AspNetCore.HealthCheck.Abstractions/CeilingThresholdWatchSettings.cs (hunk 2 above)
Comment (ycrumeyrolle) [reply src/AspNetCore.HealthCheck.Abstractions/CeilingThresholdWatchSettings.cs]: Agree

File: src/AspNetCore.HealthCheck.Abstractions/CeilingThresholdWatchSettings.cs (hunk 1 above)
Comment (ycrumeyrolle) [reply src/AspNetCore.HealthCheck.Abstractions/CeilingThresholdWatchSettings.cs]: Agree

File: src/AspNetCore.HealthCheck.Counter/CounterWatcher.cs (hunk 3 above)
Comment (ycrumeyrolle) [reply src/AspNetCore.HealthCheck.Counter/CounterWatcher.cs]: Added both warning & error threshold

**Review by sdib — APPROVED** [review]

**Review by sdib — APPROVED** [review]

**Pull request merged**